crates/solver/src/settlement/settlement_encoder.rs

```diff
@@ -373,7 +373,7 @@ impl SettlementEncoder {
         });
     }
 
-    pub fn finish(mut self) -> EncodedSettlement {
+    pub fn finish(mut self, internalize_interactions: bool) -> EncodedSettlement {
```

Comment:
Have you considered introducing an `enum` here:
```rust
enum Interactions {
    All,
    No internalized,
}
```

While this is, essentially a boolean value, when looking at the PR, I found that I had to look at the function signature to understand what the difference between `finish(true)` and `finish(false)`, and it wasn't immediately obvious at the call site.

Not super attached to this idea though, and I think it's fine as is.